docs/help/hostname_resolution.md

```diff
@@ -0,0 +1,164 @@
+🐳  Preparing Kubernetes v1.28.3 on Docker 24.0.7 ...
+    ▪ Generating certificates and keys ...
+    ▪ Booting up control plane ...
+    ▪ Configuring RBAC rules ...
+🔗  Configuring bridge CNI (Container Networking Interface) ...
+    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
+🔎  Verifying Kubernetes components...
+🌟  Enabled addons: storage-provisioner, default-storageclass
+🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
+```
+
+Get minikube's ip
+
+```console
+minikube ip
+192.168.39.54
+```
+
+Note that every time you restart minikube you'll have to follow update the ip in
+the configuration files.
+
+## /etc/hosts
+
+The generic way to make the url resolve to an IP address is to update the
+`/etc/hosts/`. There's no wildcard in this file, this means you'll need to
+specify all the urls you are interested in.
+
+Here's an example for an ingress:
+```console
+$ cat /etc/hosts
+192.168.39.54      ing.sslacceptor.send-receive-0.send-receive-project.demo.artemiscloud.io
+```
+### pros
+
+* Works on every setup and is simple
+
+### cons
+
+* No wildcard, you need to list every domains and subdomains you'll need to access
+
+## NetworkManager's DNSMasq plugin
+
+We will use networkmanager's dnsmasq plugin
+([source](https://fedoramagazine.org/using-the-networkmanagers-dnsmasq-plugin/))to
+configure the ip associated with the domain `demo.artemiscloud.io`. The dnsmasq plugin
+has wildcards which is better than setting manually every hosts in the
+`/etc/hosts` file.
+
+### Configure DNSMasq
+
+The goal here is to set up enable dnsmasq and to make it resolve your cluster
+domain `demo.artemiscloud.io` to the cluster's ip address. Because DNSMasq has a
+wildcard, all subdomains will also resolve to the same ip address.
+
+1. Create the following files:
+
+```console
+$ cat << EOF > /etc/NetworkManager/conf.d/00-use-dnsmasq.conf
+[main]
+dns=dnsmasq
+EOF
+```
+
+```console
+$ cat << EOF > /etc/NetworkManager/dnsmasq.d/00-demo.artemiscloud.io.conf
+local=/demo.artemiscloud.io/
+address=/.demo.artemiscloud.io/192.168.39.54
+EOF
+```
+
+```console
+$ cat << EOF > /etc/NetworkManager/dnsmasq.d/02-add-hosts.conf
+addn-hosts=/etc/hosts
+EOF
+```
+
+2. restart NetworkManager:
+
+```console
+$ sudo systemctl restart NetworkManager
+```
+
+### pros
+
+* Has wildcard, you only need to setup the ip once
+
+### Cons
+
+* Works only with NetworkManager
+
+## Minikube's `ingress-dns` plugin:
+
+[Follow the official documentation.](https://minikube.sigs.k8s.io/docs/handbook/addons/ingress-dns/)
+
+### pros
+
+* Has wildcard, you only need to setup the ip once
+* Supported for every setup (linux, mac, windows)
+
+### cons
+
+* Can only resolves the ingresses URLs
+
+#### caveat
+
+When using the `ingress-dns` plugin, the only urls that will get resolved are
+the ingresses. Not the other services like the Broker's acceptors.
+
+In turns this means that you need to configure `ingressHost` of the `acceptor`
```

Comment:
I am not following this explanation

Reply:
I've been running the test once again, here's what's happening:

I use the url of the ingress:  `tcp://ing.sslacceptor.send-receive-0.send-receive-project.demo.artemiscloud.io:443`
But then, when the client wants to start sending messages to the broker it uses the endpoint of the `acceptor`:  `send-receive-ss-0.send-receive-hdls-svc.send-receive-project.svc.demo.artemiscloud.io`

See the stack:
```
./artemis check queue --name TEST --produce 10 --browse 10 --consume 10 --url 'tcp://ing.sslacceptor.send-receive-0.send-receive-project.demo.artemiscloud.io:443?sslEnabled=true&trustStorePath=/tmp/broker.pem&trustStoreType=PEM' --verbose
Executing org.apache.activemq.artemis.cli.commands.check.QueueCheck check queue --name TEST --produce 10 --browse 10 --consume 10 --url tcp://ing.sslacceptor.send-receive-0.send-receive-project.demo.artemiscloud.io:443?sslEnabled=true&trustStorePath=/tmp/broker.pem&trustStoreType=PEM --verbose 
Home::/home/tlavocat/dev/activemq-artemis/artemis-distribution/target/apache-artemis-2.34.0-SNAPSHOT-bin/apache-artemis-2.34.0-SNAPSHOT, Instance::null
Connection brokerURL = tcp://ing.sslacceptor.send-receive-0.send-receive-project.demo.artemiscloud.io:443?sslEnabled=true&trustStorePath=/tmp/broker.pem&trustStoreType=PEM
Running QueueCheck
Checking that a producer can send 10 messages to the queue TEST ... 16:38:26.396 [pool-2-thread-1] ERROR org.apache.activemq.artemis.core.client - AMQ214016: Failed to create netty connection
java.net.UnknownHostException: send-receive-ss-0.send-receive-hdls-svc.send-receive-project.svc.demo.artemiscloud.io
	at java.base/java.net.InetAddress$CachedAddresses.get(InetAddress.java:797) ~[?:?]
	at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1533) ~[?:?]
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1386) ~[?:?]
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1307) ~[?:?]
	at java.base/java.net.InetAddress.getByName(InetAddress.java:1257) ~[?:?]
	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:156) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:153) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at java.base/java.security.AccessController.doPrivileged(Native Method) ~[?:?]
	at io.netty.util.internal.SocketUtils.addressByName(SocketUtils.java:153) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:41) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:61) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:53) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:55) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:31) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.AbstractAddressResolver.resolve(AbstractAddressResolver.java:106) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.bootstrap.Bootstrap.doResolveAndConnect0(Bootstrap.java:220) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.bootstrap.Bootstrap.access$000(Bootstrap.java:47) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:189) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:175) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:625) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:105) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetSuccess(AbstractChannel.java:990) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:516) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:429) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:486) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:416) ~[netty-transport-classes-epoll-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at org.apache.activemq.artemis.utils.ActiveMQThreadFactory$1.run(ActiveMQThreadFactory.java:118) ~[artemis-commons-2.34.0-SNAPSHOT.jar:2.34.0-SNAPSHOT]
success
Checking that a consumer can browse 10 messages from the queue TEST ... 16:38:28.592 [pool-2-thread-1] ERROR org.apache.activemq.artemis.core.client - AMQ214016: Failed to create netty connection
java.net.UnknownHostException: send-receive-ss-0.send-receive-hdls-svc.send-receive-project.svc.demo.artemiscloud.io
	at java.base/java.net.InetAddress$CachedAddresses.get(InetAddress.java:797) ~[?:?]
	at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1533) ~[?:?]
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1386) ~[?:?]
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1307) ~[?:?]
	at java.base/java.net.InetAddress.getByName(InetAddress.java:1257) ~[?:?]
	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:156) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:153) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at java.base/java.security.AccessController.doPrivileged(Native Method) ~[?:?]
	at io.netty.util.internal.SocketUtils.addressByName(SocketUtils.java:153) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:41) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:61) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:53) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:55) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:31) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.AbstractAddressResolver.resolve(AbstractAddressResolver.java:106) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.bootstrap.Bootstrap.doResolveAndConnect0(Bootstrap.java:220) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.bootstrap.Bootstrap.access$000(Bootstrap.java:47) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:189) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:175) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:625) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:105) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetSuccess(AbstractChannel.java:990) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:516) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:429) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:486) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:416) ~[netty-transport-classes-epoll-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at org.apache.activemq.artemis.utils.ActiveMQThreadFactory$1.run(ActiveMQThreadFactory.java:118) ~[artemis-commons-2.34.0-SNAPSHOT.jar:2.34.0-SNAPSHOT]
success
Checking that a consumer can consume 10 messages from the queue TEST ... 16:38:30.653 [pool-2-thread-1] ERROR org.apache.activemq.artemis.core.client - AMQ214016: Failed to create netty connection
java.net.UnknownHostException: send-receive-ss-0.send-receive-hdls-svc.send-receive-project.svc.demo.artemiscloud.io
	at java.base/java.net.InetAddress$CachedAddresses.get(InetAddress.java:797) ~[?:?]
	at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1533) ~[?:?]
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1386) ~[?:?]
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1307) ~[?:?]
	at java.base/java.net.InetAddress.getByName(InetAddress.java:1257) ~[?:?]
	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:156) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:153) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at java.base/java.security.AccessController.doPrivileged(Native Method) ~[?:?]
	at io.netty.util.internal.SocketUtils.addressByName(SocketUtils.java:153) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:41) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:61) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:53) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:55) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:31) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.resolver.AbstractAddressResolver.resolve(AbstractAddressResolver.java:106) ~[netty-resolver-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.bootstrap.Bootstrap.doResolveAndConnect0(Bootstrap.java:220) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.bootstrap.Bootstrap.access$000(Bootstrap.java:47) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:189) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:175) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:625) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:105) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetSuccess(AbstractChannel.java:990) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:516) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:429) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:486) ~[netty-transport-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:416) ~[netty-transport-classes-epoll-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.108.Final.jar:4.1.108.Final]
	at org.apache.activemq.artemis.utils.ActiveMQThreadFactory$1.run(ActiveMQThreadFactory.java:118) ~[artemis-commons-2.34.0-SNAPSHOT.jar:2.34.0-SNAPSHOT]

```

Reply:
`Checks run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 6.369 sec - QueueCheck
`

Reply:
it think that is the core protocol using the url from the topology that is returned as part of core discovery, or at least it looks like that. It is only the broker that will know about that dns name.

one way to verify is to use --protocol=AMQP on the artemis check command, it has some awareness of topology but it may be easier to disable. I need to find the reference in the doc that talks about disabling topology updates. on way is to disable the cluster connection I think.

Reply:
it may be as simple as adding these options to the urls:

**AMQP: failover.amqpOpenServerListAction=IGNORE
CORE: useTopologyForLoadBalancing=false**

Reply:
for success, maybe if the topology info fails it reverts to the host url? if you can configure logging for the client and enable debug/trace logging it may be come clear.

Reply:
Ah yes indeed! It was trying to use the topology for load balancing, which must be disabled since we're connecting through an ingress and the load balancing is done using a round robin algorithm.

Here's the output with the `useTopologyForLoadBalancing=false`:

```
./artemis check queue --name TEST --produce 10 --browse 10 --consume 10 --url 'tcp://ing.sslacceptor.send-receive-0.send-receive-project.demo.artemiscloud.io:443?sslEnabled=true&trustStorePath=/tmp/IssuerCA.pem&trustStoreType=PEM&useTopologyForLoadBalancing=false'
Connection brokerURL = tcp://ing.sslacceptor.send-receive-0.send-receive-project.demo.artemiscloud.io:443?sslEnabled=true&trustStorePath=/tmp/IssuerCA.pem&trustStoreType=PEM&useTopologyForLoadBalancing=false
Running QueueCheck
Checking that a producer can send 10 messages to the queue TEST ... success
Checking that a consumer can browse 10 messages from the queue TEST ... success
Checking that a consumer can consume 10 messages from the queue TEST ... success
Checks run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.325 sec - QueueCheck
```

Reply:
I think we've got all we need to wrap up the tutorial, I'll update it with these latest findings and will send it back for review this morning. Thanks for your help!